components/DisplayDonatedItems.tsx

```diff
@@ -0,0 +1,74 @@
+'use client';
+
+import { getItems } from '@/supabase/models/getItems';
+import Link from 'next/link';
+import Modal from '@/components/Modal';
+import ItemCard from '@/components/ItemCard';
+import { useEffect, useState } from 'react';
+import { PartialItem } from '@/types/supabaseTypes';
+import useMediaQuery from './hooks/useMediaQuery';
+
+type DisplayDonatedItemsProps = {
+  userId: string;
+};
+
+const DisplayDonatedItems = ({ userId }: DisplayDonatedItemsProps) => {
+  const isBreakpoint = useMediaQuery(1000);
+  const [storeItems, setStoreItems] = useState<PartialItem[]>([]);
+
+  useEffect(() => {
+    const fetchItems = async () => {
+      try {
+        const fetchItemsRes = await getItems('items', '', 'donated_by', userId);
+        setStoreItems(fetchItemsRes || []);
```

Comment:
```suggestion
        const itemsResult = await getItems('items', '', 'donated_by', userId);
        itemsResult && setStoreItems(itemsResult);
```
Don't run away from an error - if we set an empty array as a fallback we'll never see an issue in the UI, which might sound great but creates a problem if we need to debug.

Always fail fast. `getItems` throws an error that will get caught in our catch and we set the state if the return value of that function is not null - meaning that if something comes back that does not fit our expected type or has any other issue we will be forced to deal with it.

Also, don't abbreviate your variable names 🔪